apps/hexdailystats/hex_daily_stats.star

```diff
@@ -0,0 +1,186 @@
+"""
+Applet: HEX Daily Stats
+Summary: HEX Daily Stats
+Description: Displays HEX price, Payout per T-Share and T-Share rate
+Author: kmphua
+Thanks: aschober, bretep, codeakk
+"""
+
+load("render.star", "render")
+load("http.star", "http")
+load("encoding/base64.star", "base64")
+load("encoding/json.star", "json")
+load("cache.star", "cache")
+load("math.star", "math")
+
+COINGECKO_PRICE_URL = "https://api.coingecko.com/api/v3/coins/{}?localization=false&tickers=false&community_data=false&developer_data=false"
+
+HEX_ICON_SM = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAAkAAAAICAYAAAArzdW1AAAACXBIWXMAAAsTAAAL
+EwEAmpwYAAABG0lEQVQYlQXBPUsCYQDA8f+dz6WXr3cqZlpkkSCBBI2NITSI0NTi
+B2jpOzT1CRqipYjmliKiJQiioWhoiYqIuDDMzJcwz7vTp99PkZUNeJw3qNkPuIko
+6hTSmwBftqPkkkV0qyUQEtTbA/yjLNok9L9QlDTSqYelld9VCoF1RVZXy1jdE37N
+bZr+OF7cxUsU+AmvIGdgLr/o26qo10THTmGQR/4to7kxDO0IpzuN3Uxh2yXBgrrD
+Z7uENooRDn7z0V9i4KjMpo55aRWRzpggIw7RB0XG3XuS/SbpyCvP1hpDLuj72hjB
+qiCuvxHqnOM19ghEwBT7xMQZ77UyunKHkbkSjDzImpeojRZtt05v0CMyfMIMpQnl
+Nrlx+AeJZmcOHgyOkAAAAABJRU5ErkJggg==
+""")
+
+NO_DATA = "---------- "
+
+HEX_SUBGRAPH_URL = "https://api.thegraph.com/subgraphs/name/codeakk/hex"
+POST_HEADERS = {
+    "Content-Type": "application/json",
+}
+DAILY_DATA_UPDATES_QUERY = "{\"query\": \"{ dailyDataUpdates(orderBy: timestamp orderDirection: desc first: 1) { payoutPerTShare } }\" }"
+GLOBAL_INFOS_QUERY = "{\"query\": \"{ globalInfos(orderBy: timestamp orderDirection: desc first: 1)  { hexDay, shareRate } }\" }"
+
+def main(config):
+    # Get coin data for selected coin from CoinGecko
+    coin_data = get_json_from_cache_or_http(COINGECKO_PRICE_URL.format("hex"), ttl_seconds = 600)
+
+    # Check for catastrophic data failure (i.e. failed to get data from CoinGecko and no cache data is available to fall back on)
+    if coin_data == None:
+        display_error = render.Column(
+            expanded = True,
+            main_align = "space_evenly",
+            cross_align = "center",
+            children = [
+                render.Text("ERROR:", font = "CG-pixel-3x5-mono", color = "#FF0000"),
+                render.Text("CoinGecko API", font = "CG-pixel-3x5-mono"),
+                render.Text("unvailable", font = "CG-pixel-3x5-mono"),
+            ],
+        )
+        print("Error: No CoinGecko data available")
+        return render.Root(
+            child = render.Box(
+                display_error,
+            ),
+        )
+
+    hex_price = str("$%f" % float(coin_data["market_data"]["current_price"]["usd"]))
+
+    # Get HEX daily data updates
+    daily_data_updates_response = http.post(HEX_SUBGRAPH_URL, body = DAILY_DATA_UPDATES_QUERY, headers = POST_HEADERS)
```

Comment:
Would it be appropriate to cache these calls? I can't say I understand the queries, but if it's daily data then a TTL of  several hours would be fine right?

Reply:
Agreed. I added an 8 hour cache for the daily data.